### ISSUE TITLE
fully cached build is slow and indicates multiple 5 second sleeps

### ISSUE THREAD

**Issue by andybrown668** [earthly/earthly#2652]
how to reproduce:
1. run earthly -V  +for-linux
2. run step 1 again 

expected result: the second invocation is very fast
actual result: the second invocation takes 31s

I have this same issue with my own builds and I notice 'context' appears to be the cause:
```
             context | Completed in 5.302829389s
             context | Completed in 5.326069956s
             context | Completed in 5.352032439s
             context | Completed in 5.381829586s
             context | Completed in 5.418959072s
             context | Completed in 5.447935549s
             context | Completed in 5.4805472s
```
 Those timings all indicate a 5 second sleep somewhere - not sure if it's buildkit or earthly

note, I tried to catch this in the debugger but I can't successfully run an earthly build in the debugger (Goland):
```
GOROOT=/home/abrown/go/go1.19.5 #gosetup
GOPATH=/home/abrown/go #gosetup
/home/abrown/go/go1.19.5/bin/go build -o /tmp/GoLand/___go_build_github_com_earthly_earthly_cmd_earthly github.com/earthly/earthly/cmd/earthly #gosetup
/tmp/GoLand/___go_build_github_com_earthly_earthly_cmd_earthly -V +for-linux
 Init 🚀
————————————————————————————————————————————————————————————————————————————————

            frontend | Running command: docker info
            frontend | Running command: docker info --format={{.SecurityOptions}}
            frontend | Running command: docker info --format={{.DockerRootDir}}
            frontend | docker-shell frontend initialized.
            frontend | Running command: docker container inspect earthly-buildkitd
           buildkitd | Found buildkit daemon as docker container (earthly-buildkitd)
            frontend | Running command: docker container inspect earthly-buildkitd
            frontend | Running command: docker image inspect 
           buildkitd | Comparing running container "earthly-buildkitd" image ("sha256:c1fe02837a933c70884d8cd73fd44a12c7740d9bddce847ff3f5197d33a1454e") with available image "" ("")
           buildkitd | Updated image available. Restarting buildkit daemon...
            frontend | Running command: docker stop -t 10 earthly-buildkitd
            frontend | Running command: docker container inspect earthly-buildkitd
            frontend | Running command: docker container inspect earthly-buildkitd
            frontend | Running command: docker rm earthly-buildkitd
            frontend | Running command: docker image inspect 
            frontend | Running command: docker run --privileged --env BUILDKIT_DEBUG=false --env BUILDKIT_TCP_TRANSPORT_ENABLED=false --env BUILDKIT_TLS_ENABLED=false --env BUILDKIT_MAX_PARALLELISM=20 --label dev.earthly.settingshash=c47eed41a88b6f3b --mount type=volume,source=earthly-cache,dst=/tmp/earthly --publish 127.0.0.1:8371:8371/tcp -d --pull missing --name earthly-buildkitd 
Share your logs with an Earthly account (experimental)! Register for one at https://ci.earthly.dev.
Error: build new buildkitd client: maybe start buildkitd: maybe restart: could not start buildkit: 1 error occurred:
        * command failed: docker run --privileged --env BUILDKIT_DEBUG=false --env BUILDKIT_TCP_TRANSPORT_ENABLED=false --env BUILDKIT_TLS_ENABLED=false --env BUILDKIT_MAX_PARALLELISM=20 --label dev.earthly.settingshash=c47eed41a88b6f3b --mount type=volume,source=earthly-cache,dst=/tmp/earthly --publish 127.0.0.1:8371:8371/tcp -d --pull missing --name earthly-buildkitd : exit status 125: docker: invalid reference format.
See 'docker run --help'.: exit status 125



Process finished with the exit code 1
```

**Comment by alexcb** [comment]
This almost sounds like #2049 (which has been fixed and released in v0.6.30)

does this occur in the latest [rc2](https://github.com/earthly/earthly/releases/tag/v0.7.0-rc2)

or if you re-build earthly using the freshly built earthly? e.g. `./build/linux/amd64/earthly -V +for-linux`?

**Comment by andybrown668** [comment]
Thanks Alex, v0.6.30 works *much* faster - problem solved!
